src/service/resource.js

```diff
@@ -152,6 +152,9 @@ export class Resource {
     /** @const @private {!./resources-impl.Resources} */
     this.resources_ = resources;
 
+    /** @const @private {!./viewport/viewport-impl.Viewport} */
+    this.viewport_ = Services.viewportForDoc(element);
```

Comment:
This might introduce a race condition. Can we just move this down into the measure method?

Reply:
which 2 can race?

Reply:
moved it down to the method.